### PR TITLE
AttributionDocumentGeneratorImpl can generate the docuement with files

### PR DESCRIPTION
**Pull request by bs-ondem**
A new functionality is added to the AttributionDocumentGeneratorImpl,
to generate the attribution document only with given template files
without the need of the ServiceLoader.

The antenna-demo_template.* PDF test files are also added to the
attribution-document-core JAR to reuse the files in attribution-
document-generator tests.

### Request Reviewer
@sschuberth @blaumeiser-at-bosch 

### Type of Change
*improvement*

### CODE REVIEW

**Comment by sschuberth** [comment]
Commit message typo: s/docuement/document/

File: modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/utils/TemplateLoaderUtil.java [hunk 1]
```diff
@@ -32,9 +33,57 @@ private TemplateLoaderUtil() {
...
     public static Templates load(String key) {
-        TemplateBundle bundle = getLoader(key);
+        return load(getLoader(key));
+    }
+
+    public static Templates load(File cover, File copyright, File content, File back) {
+        return load(new TemplateBundle() {
+            @Override
+            public String key() {
+                return "";
```
Comment (sschuberth) [inline modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/utils/TemplateLoaderUtil.java:43]: Would it make sense to derive an "artificial" key from the files names? I'm not sure how the return value of this function is used, if at all. Maybe it's just fine to always return a empty string here.

File: modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/utils/TemplateLoaderUtil.java [hunk 2]
```diff
@@ -32,9 +33,57 @@ private TemplateLoaderUtil() {
...
+                } catch (IOException e) {
+                    throw new ExecutionException("Could not load back template PDF", e);
+                }
+            }
+        });
+    }
+
+    public static Templates load(TemplateBundle bundle) {
         Templates templates = new Templates();
+
```
Comment (sschuberth) [inline modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/utils/TemplateLoaderUtil.java:86]: Nit: Unrelated formatting change.

File: modules/attribution-document/attribution-document-generator/src/test/java/org/eclipse/sw360/antenna/attribution/document/workflow/generators/YagenIntegrationTest.java [hunk 3]
```diff
@@ -31,25 +34,52 @@
...
+              new AttributionDocumentGeneratorImpl("attribution_document.pdf", tmpDir, "antenna-demo", values);
 
       List<ArtifactAndLicense> foo = createData();
       File file = attrDocGen.generate(foo);
       assertThat(file).exists().hasExtension("pdf");
    }
 
+   @Test
+   void goGenerateWithFiles() throws IOException {
+      int expectedResultSize = 189490;
```
Comment (sschuberth) [inline modules/attribution-document/attribution-document-generator/src/test/java/org/eclipse/sw360/antenna/attribution/document/workflow/generators/YagenIntegrationTest.java:59]: Maybe again inline this (as "pdf" below is also not put into a variable, and it makes more clear at a single glance what the expected file size is)?

File: modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/utils/TemplateLoaderUtil.java [hunk 4]
```diff
@@ -32,9 +33,57 @@ private TemplateLoaderUtil() {
...
+        return load(new TemplateBundle() {
+            @Override
+            public String key() {
+                return "";
+            }
+
+            @Override
+            public InputStream loadTitleTemplate() {
+                try {
+                    return new FileInputStream(cover);
```
Comment (sschuberth) [inline modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/utils/TemplateLoaderUtil.java:49]: Where are these streams closed? I couldn't find code / a caller that closes these.

File: modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/utils/TemplateLoaderUtil.java (hunk 1 above)
Comment (bs-ondem) [reply modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/utils/TemplateLoaderUtil.java:43]: Actually it's fine to return only an empty string in this case, because the key is only used when loading the template via the ServiceLoader.

File: modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/utils/TemplateLoaderUtil.java (hunk 4 above)
Comment (bs-ondem) [reply modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/utils/TemplateLoaderUtil.java:49]: These are closed here: https://github.com/eclipse/antenna/blob/master/modules/attribution-document/attribution-document-core/src/main/java/org/eclipse/sw360/antenna/attribution/document/core/Templates.java#L115-L141
